src/methods/clipping/coverage.jl

```diff
@@ -0,0 +1,301 @@
+export coverage
+
+#=
+## What is coverage?
+
+Coverage is the amount of geometry area within a bounding box defined by the minimum and
+maximum x and y-coordiantes of that bounding box, or an Extent containing that information.
+
+To provide an example, consider this rectangle:
+```@example rect
+import GeometryOps as GO
+import GeoInterface as GI
+using Makie
+using CairoMakie
+
+rect = GI.Polygon([[(-1,0), (-1,1), (1,1), (1,0), (-1,0)]])
+cell = GI.Polygon([[(0, 0), (0, 2), (2, 2), (2, 0), (0, 0)]])
+xmin, xmax, ymin, ymax = 0, 2, 0, 2
+f, a, p = poly(collect(GI.getpoint(cell)); axis = (; aspect = DataAspect()))
+poly!(collect(GI.getpoint(rect)))
+f
+```
+It is clear that half of the polygon is within the cell, so the coverage should be 1.0, half
+of the area of the rectangle. 
+```@example rect
+GO.coverage(rect, xmin, xmax, ymin, ymax)
+```
+
+## Implementation
+
+This is the GeoInterface-compatible implementation. First, we implement a wrapper method
+that dispatches to the correct implementation based on the geometry trait. This is also used
+in the implementation, since it's a lot less work!
+
+Note that the coverage is zero for all points and curves, even if the curves are closed like
+with a linear ring.
+=#
+
+# Targets for applys functions
+const _COVERAGE_TARGETS = Union{GI.PolygonTrait,GI.AbstractCurveTrait,GI.MultiPointTrait,GI.PointTrait}
```

Comment:
Does coverage make sense for points or multi points?

Reply:
I did what we did for `area` and it just returns zero for anything that isn't a polygon. I think it shouldn't error if you do supply in something other than a polygon, especially since you don't always know what types you're working with when some geometry functions return (for example LibGEOS functions).

Reply:
Makes sense.